server/src/controllers/course.ts

```diff
@@ -40,3 +62,161 @@ export async function addCourse(req: Request, res: Response): Promise<void> {
     });
   }
 }
+
+export async function getCourse(req: Request, res: Response): Promise<Response> {
+  try {
+    const courseId: number = Number(req.params.courseId);
+    await idSchema.validate({ id: courseId });
+    const course: courseService.CourseWithTranslation = await courseService.findCourseById(courseId);
+  
+    const courseData: CourseData = {
+      id: course.id,
+      courseCode: course.courseCode,
+      minCredits: course.minCredits,
+      maxCredits: course.maxCredits,
+      department: {
+        en: '',
+        fi: '',
+        sv: ''
+      },
+      name: {
+        en: '',
+        fi: '',
+        sv: ''
+      },
+      evaluationInformation: {
+        en: '',
+        fi: '',
+        sv: ''
+      }
+    };
+  
+    course.CourseTranslations.forEach((translation: CourseTranslation) => {
+      switch (translation.language) {
+      case Language.English:
+        courseData.department.en = translation.department;
+        courseData.name.en = translation.courseName;
+        break;
+      case Language.Finnish:
+        courseData.department.fi = translation.department;
+        courseData.name.fi = translation.courseName;
+        break;
+      case Language.Swedish:
+        courseData.department.sv = translation.department;
+        courseData.name.sv = translation.courseName;
+        break;
+      }
+    });
+  
+    return res.status(200).send({
+      success: true,
+      course: courseData
+    });
+  } catch (error: unknown) {
+    console.log(error);
+
+    if (error instanceof yup.ValidationError) {
+      return res.status(400).send({
+        success: false,
+        error: error.errors
+      });
+    }
+
+    if (error instanceof Error && error?.message.startsWith('course with an id')) {
```

Comment:
In the long run, it could be useful to have our own Error classes, for example `CourseNotFoundError`. Then, we could use `instanceof` here, too.

Reply:
100% agree, it will make implementing the error handler much easier once we have proper error classes. We are just starting to see what kind of errors we will encounter (apart from the obvious ones).